src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -188,11 +266,27 @@ public void messageReceived(BytesTransportRequest request, final TransportChanne
                 in = request.bytes().streamInput();
             }
             in.setVersion(request.version());
-            ClusterState clusterState = ClusterState.Builder.readFrom(in, nodesProvider.nodes().localNode());
-            clusterState.status(ClusterState.ClusterStateStatus.RECEIVED);
-            logger.debug("received cluster state version {}", clusterState.version());
+            lock.lock();
```

Comment:
if we only use the lock can't se just use synchronized(this) {} instead and we safe the try block?